tests/cli/test_list.py

```diff
@@ -276,4 +331,398 @@ def test_list_json_with_circular_reverse(project, capsys, repository):
+    foo_l = _MockPackagePath("foo-0.1.0.dist-info", "LICENSE")
+    foo_l.license_text = "license text for foo here"
+    foo.files = [foo_l]
+
+    # Bar package.
+    bar = Distribution("bar", "3.0.1", metadata={"License": "B License"})
+    bar_l = _MockPackagePath("bar-3.0.1.dist-info", "LICENSE")
+    bar_l.license_text = "license text for bar here"
+    bar.files = [bar_l]
+
+    # Baz package.
+    baz = Distribution("baz", "2.7", metadata={"License": "C License"})
+    baz_l = _MockPackagePath("bar-2.7.dist-info", "LICENSE")
+    baz_l.license_text = "license text for baz here"
+    baz.files = [baz_l]
+
+    # missing package- License is set to UNKNOWN, text saved in COPYING
+    unknown = Distribution(
+        "unknown",
+        "1.0",
+        metadata={
+            "License": "UNKNOWN",
+            "Classifier": "License :: OSI Approved :: Apache Software License",
+        },
+    )
+    unknown_l = _MockPackagePath("unknown-1.0.dist-info", "COPYING")
+    unknown_l.license_text = "license text for UNKNOWN here"
+    unknown.files = [unknown_l]
+
+    # missing package- License is set to UNKNOWN, text saved in LICENCE
+    # using UK spelling
+    classifier = Distribution(
+        "classifier", "1.0", metadata={"Classifier": "License :: PDM TEST D"}
+    )
+    classifier_l = _MockPackagePath("classifier-1.0.dist-info", "LICENCE")
+    classifier_l.license_text = "license text for CLASSIFIER here"
+    classifier_l.read_text = lambda *a, **kw: 1 / 0  # make it throw an error
+    classifier.files = [classifier_l]
+
+    # Place our fake packages in the working set.
+    for candidate in [foo, bar, baz, unknown, classifier]:
+        working_set.add_distribution(candidate)
+
+
+@mock.patch("pdm.termui.ROUNDED", ASCII)
+@pytest.mark.usefixtures("working_set")
+def test_list_bare_sorted_version(project, invoke):
+    actions.do_add(project, packages=["requests"])
+    result = invoke(["list", "--sort", "version"], obj=project)
+    expected = (
+        "+--------------------------------------+\n"
+        "| name         | version    | location |\n"
+        "|--------------+------------+----------|\n"
+        "| test-project | 0.0.0      |          |\n"
+        "| urllib3      | 1.22       |          |\n"
+        "| requests     | 2.19.1     |          |\n"
+        "| idna         | 2.7        |          |\n"
+        "| certifi      | 2018.11.17 |          |\n"
+        "| chardet      | 3.0.4      |          |\n"
+        "+--------------------------------------+\n"
+    )
+    assert expected == result.output
+
+
+# # TODO: resolve with graph?
+# # TODO: how to fix this?
+# # "[CandidateNotFound]: No candidate is found for `requests` that
+# # matches the environment or hashes\nAdd '-v' to see the detailed traceback\n"
+# @pytest.mark.usefixtures("working_set", "repository")
+# def test_list_bare_sorted_version_resolve(project, invoke, repository):
+#     # project.environment.python_requires = PySpecSet(">=3.6")
```

Comment:
Shouldn't this line fix the `CandidateNotFound` issue?

Reply:
I thought so, but I cannot make it work.  Do you have any ideas?

Reply:
I tried quite a lot of different things here, and I am not sure what to try next.  I would appreciate any advice :)

Reply:
You need to mock the `.metadata` cuz it will be accessed during the command, which will try to find a package link and fail.
https://github.com/pdm-project/pdm/blob/cd2f1a98513d03f03fcb784804b9eff10fb37871/src/pdm/cli/commands/list.py#L144-L146

Reply:
I have not worked out how to mock this yet.  How can I access and mock the candidates within the test fixture?